main.cpp

```diff
@@ -0,0 +1,3728 @@
+#include <SDL.h>
+#include <math.h>
+#include <assert.h>
+#include <vector>
+#include <algorithm>
+#include "gl2/Context.h"
+#include <demo-config.h>
+
+static const float Pi = 3.14159265358979323846f;
+
+
+//---------------------------------------------------------------------------
+//  Data structures for the 3D model defined at the end of this file
```

Comment:
What is this file? Did you just copy and paste a code file from somewhere? This file is massive. Could you just use a basic main file that just opens a regular window and that's it?